homeassistant/helpers/llm.py

```diff
@@ -884,3 +894,48 @@ async def async_call(
         ]
 
         return {"success": True, "result": events}
+
+
+class WeatherForecastTool(Tool):
+    """LLM Tool wrapper for weather forecast action."""
+
+    name = f"{WEATHER_DOMAIN}_{SERVICE_GET_FORECASTS}"
+    description = "Get weather forecasts"
+    parameters = vol.Schema(
+        {
+            vol.Required("type"): vol.In(("daily", "hourly", "twice_daily")),
+            vol.Optional(ATTR_NAME, description="Weather entity name"): cv.string,
```

Comment:
I noticed that for calendar, just limiting this to the actual allowed weather values gives a lot better results.

Reply:
This makes sense, smart homes don't usually have too many weather entities. I will do it.